session/sshforward/sshprovider/raw_provider.go

```diff
@@ -0,0 +1,79 @@
+package sshprovider
+
+import (
+	"context"
+	"net"
+
+	"github.com/moby/buildkit/session"
+	"github.com/moby/buildkit/session/sshforward"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type rawConfig struct {
+	ID     string
+	Dialer func(context.Context) (net.Conn, error)
+}
+
+func newRawProvider(confs []rawConfig) (session.Attachable, error) {
```

Comment:
If `rawConfig` renamed to dialer as described in other comment then this can just be `new SSHProvider` or `newSocketProvider`.